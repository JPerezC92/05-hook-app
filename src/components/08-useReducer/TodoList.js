import React from "react";
import TodoListItem from "./TodoListItem";
import PropTypes from "prop-types";

const TodoList = ({ todos, handleToggle, handleDelete }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map((todo, i) => (
        <TodoListItem
          key={todo.id}
          {...todo}
          i={i}
          handleToggle={handleToggle}
          handleDelete={handleDelete}
        />
      ))}
    </ul>
  );
};
TodoList.propTypes = {
  todos: PropTypes.arrayOf(PropTypes.object),
  handleToggle: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};

export default TodoList;
