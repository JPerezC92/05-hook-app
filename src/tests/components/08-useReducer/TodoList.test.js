import { shallow } from "enzyme";
import TodoList from "../../../components/08-useReducer/TodoList";
import { demoTodos } from "../../Fixtures/demoTodos";

describe("Pruebas en el componente <TodoList />", () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
    />
  );
  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener 2 <TodoListItem />", () => {
    const TodoListItem = wrapper.find("Memo()").at(0);

    expect(wrapper.find("Memo()").length).toBe(demoTodos.length);
    expect(TodoListItem.prop("handleDelete")).toEqual(expect.any(Function));
    expect(TodoListItem.prop("handleToggle")).toEqual(expect.any(Function));
  });
});
