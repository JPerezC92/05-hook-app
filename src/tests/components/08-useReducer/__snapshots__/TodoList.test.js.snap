// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <TodoList /> Debe de mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <Memo()
    description="Aprender React"
    done={false}
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    id={1}
    key="1"
  />
  <Memo()
    description="Aprender Angular"
    done={false}
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    id={2}
    key="2"
  />
</ul>
`;
